src/components/HighTable/HighTable.tsx

```diff
@@ -436,22 +462,25 @@ export default function HighTable({
                       showCheckBox={showSelection}
                       ariaColIndex={1}
                     >{formatRowNumber(dataIndex)}</RowHeader>
-                    {data.header.map((column, columnIndex) => {
-                      // Note: the resolved cell value can be undefined
-                      const hasResolved = column in row.cells
-                      const value = row.cells[column]
-                      return <Cell
-                        key={columnIndex}
-                        onDoubleClick={getOnDoubleClickCell(columnIndex, dataIndex)}
-                        onMouseDown={getOnMouseDownCell(columnIndex, dataIndex)}
-                        stringify={stringify}
-                        value={value}
-                        columnIndex={columnIndex}
-                        hasResolved={hasResolved}
-                        className={columnClassNames[columnIndex]}
-                        ariaColIndex={columnIndex + 2} // 1-based index, +1 for the row header
-                      />
-                    })}
+                    {data.header
+                      .map((column, index) => ({ column, index }))
+                      .filter(({ index }) => isColumnVisible(index))
+                      .map(({ column, index: columnIndex }) => {
```

Comment:
At first I was going to comment on this map/filter/map seems redundant. But as I was writing it I understood that you need the index to be stable even after the filter. So you're right. But maybe add a comment?